src/loading.jl

```diff
@@ -26,13 +42,15 @@ If the desired device is CUDA (as determined by `ClimaComms.device()`), try load
 """
 macro import_required_backends()
     return quote
-        @static if $mpi_is_required() && $mpi_ext_is_not_loaded()
+        @static if $mpi_is_required() && $(!mpi_ext_is_loaded())
             import MPI
-            @info "Loaded MPI.jl"
+            @debug "Loaded MPI via `import MPI`"
+            @assert $(mpi_ext_is_loaded()) "Importing MPI did not succesfully load the ClimaCommsMPIExt"
         end
-        @static if $cuda_is_required() && $cuda_ext_is_not_loaded()
+        @static if $cuda_is_required() && $(!cuda_ext_is_loaded())
             import CUDA
-            @info "Loaded CUDA.jl"
+            @debug "Loaded CUDA via `import CUDA`"
+            @assert $(cuda_ext_is_loaded()) "Importing CUDA did not succesfully load the ClimaCommsCUDAExt"
```

Comment:
Interesting @Sbozzolo, this is failing. How did you previously test this?

I'm feeling ever more inclined to ditch this and move forward with #78. This is not worth the headache.

Also, yielding code loading to users, for example, makes this code compatible with `Distributed` (where code loading needs things like `Distributed.@everywhere import CUDA`).

Reply:
This was tested in buildkite by running the various cases. For example, 
https://buildkite.com/clima/climacomms-ci/builds/251#018f279a-7738-4bf9-9b4c-7591659f0a2d shows that CUDA was loaded and the test passed, meaning that the extension was loaded too. You can also verify this by adding a print statement in the extension. 

It might that the the test is failing for the same reason things fail in ClimaAtmos, it has to do with macro expansions and the `@static` call, which behaves differently from normal macros in the rules on how things are expanded and evaluated. However, I don't see this immediately and I think this should work.

We can go ahead with #78, but I think it would still be useful to keep the `import_required_backends` macro around at least for one minor release. We ported almost all our repos and the repos that we have transitioned would not see any advantage in changing how the import is done at the moment.